index.html

```diff
@@ -274,50 +294,67 @@ <h3>Namespace Specific Identifier (NSI)</h3>
-                a length of either 21 or 22 base58 characters. The encoding of <code>idstring</code> uses most alphas and digits, 
-                omitting 0 (zero), O (capital o), I (capital i), and l (lower L) to avoid readability problems. Regardless of 
-                the length of <code>idstring</code>, peer DIDs are case-sensitive and may not be case-normalized, even though 
-                the prefix is always lower-case.</p>
-              
-                <p>Note that this spec may be updated to include other <code>keyfmtchar</code> variants for
-                SecP256, Curve448, or similar. Such updates might change the length of <code>idstring</code> as well.</p>
+              Peer DIDs use an underlying number with high entropy, called the <em>numeric basis</em>, as the
+                source of their uniqueness. This version of
+                the spec only defines one scheme for generating the numeric basis (<code>generator</code>
+                == <code>1</code>, which is to use a hash of the initial content of a DID Doc (see next section).
+                Additional generation methods may be added in the future.</p>
+              <p>This <em>numeric basis</em> is encoded as a hash using the
+                  <a target="_blank" href="https://github.com/multiformats/multihash">multihash</a> prefix
+                  for sha256+hex: "1220". Although this prefix is redundant today, a multihash prefix is used
+                  so that a different digest or encoding may be selected in the future.</p>
+              <p>The multihash value is preceded by a <a target="_blank"
+                    href="https://github.com/multiformats/multibase">multibase prefix</a> that tells how to
+                  interpret the multihash that follows. Since the multihash value is hex in
+                  this version of the spec, <code>multibase</code> is always "F" today; if the
+                  multihash variant changes in the future, <code>multibase</code> probably will, too.<p>
+              <p>Peer DIDs must be compared according to whatever case-sensitivity is implied by the
+              hash encoding they use. Hex is not case-sensitive, but if a future version of the spec
+              uses base58 or base64 encoding, values will be case sensitive. Therefore, it is strongly
+              recommended that routines that handle peer DIDs not normalize case unless they take into
+              account the specific hash encoding.</p>
+              <p>As with git commit hashes, it is likely that a small subset of the bits in the <em>numeric basis</em>
+                  of a peer DID will be sufficiently unique for some contexts, and will be more convenient for human
+                  interaction. Therefore, a lossy <em>short form</em> of a peer DID is recognized if the hash encoding
+                  is hex. This form must consist of at least 4 bytes (8 hex characters) after the multibase and
```

Comment:
I'm a little nervous about recommending 4 byte short forms since that would be relatively easy to brute force and might cause trouble if systems have implicit assumptions about these being unique.

Reply:
These short forms are not intended to carry a strong uniqueness guarantee--only to be "unique enough" that they could be used in contexts where a human is involved (e.g., when inspecting a log or checking a DID value on a screen). They have the same uniqueness property as short git commit hashes. Git has chosen 7 hex digits as a "good enough" uniqueness, and I was trying to do something similar.

I introduced this concept because I wanted the simplicity of hex instead of base58 or base64 or something--but I didn't want the length to be a problem for humans, and I was abandoning the idea of shortening the SHA256 hash to just the top half of its value, so the long form of the DID would be quite verbose. If we don't like this short form, then I'd vote to go back to base58.

Does that reasoning change your opinion in any way?

Reply:
@christianlundkvist I'm thinking through your comment about brute forcing. In a log file, I don't think brute forcing introduces much risk, but if we're talking about reading off the screen, maybe. I'm pondering...

Reply:
What if we said the short form of the DID was the first 4 hex digits of the longer hex value, followed by "." followed by 4 hex digits of CRC of the value as a whole? This would require brute forcing to find a different doc that had the same sha256 value (extremely unlikely), or a different doc that had a sha256 value with the same 4 initial hex digits, plus the same CRC (fairly unlikely, but not as unlikely as a sha256 collision). Would this allow the short form while preventing the attack?

Reply:
Never mind. That only gives 1-in-4-billion security. Still easy to brute force.